zone.go

```diff
@@ -231,6 +231,31 @@ type PurgeCacheResponse struct {
 	} `json:"result"`
 }
 
+// ZoneSubscriptionDetails represents all the subscription details for the zone.
+type ZoneSubscriptionDetails struct {
+	ID       string `json:"id,omitempty"`
+	State    string `json:"state,omitempty"`
+	Price    int    `json:"price,omitempty"`
+	Currency string `json:"currency,omitempty"`
+	Zone     Zone   `json:"zone,omitempty"`
+	RatePlan struct {
+		ID                string `json:"id,omitempty"`
+		Name              string `json:"public_name,omitempty"`
+		Currency          string `json:"currency,omitempty"`
+		ExternallyManaged bool   `json:"externally_managed,omitempty"`
+		Scope             string `json:"scope,omitempty"`
+	} `json:"rate_plan,omitempty"`
+	Frequency          string    `json:"frequency,omitempty"`
+	CurrentPeriodStart time.Time `json:"current_period_end,omitempty"`
```

Comment:
The key name doesn't match the correct JSON field. You'll need to swap these two around so that the start reads from the start JSON field.